MdeModulePkg/Core/Dxe/Misc/MemoryAttributesTable.c

```diff
@@ -395,11 +395,14 @@ MergeMemoryMap (
   NewMemoryMapEntry = MemoryMap;
   MemoryMapEnd      = (EFI_MEMORY_DESCRIPTOR *)((UINT8 *)MemoryMap + *MemoryMapSize);
   while ((UINTN)MemoryMapEntry < (UINTN)MemoryMapEnd) {
-    CopyMem (NewMemoryMapEntry, MemoryMapEntry, sizeof (EFI_MEMORY_DESCRIPTOR));
+    CopyMem (NewMemoryMapEntry, MemoryMapEntry, DescriptorSize);

```

Comment:
Is there the condition that DescriptorSize is not equal to sizeof (EFI_MEMORY_DESCRIPTOR)？

Reply:
If the final entry is at the end of a page with no valid page following it, then this can cause an access violation.

Reply:
I mean why use DescriptorSize to replace sizeof (EFI_MEMORY_DESCRIPTOR)?

Reply:
I'm not sure if there's a case where `DescriptorSize` is not equal to `sizeof (EFI_MEMORY_DESCRIPTOR)` but I think it makes more senes to use `DescriptorSize` here.

The API describes `DescriptorSize` as the size in bytes of an individual EFI_MEMORY_DESCRIPTOR which logically makes sense to use for the CopyMem call.   Additionally, in the next line we reference the size of the same variable in the `NEXT_MEMORY_DESCRIPTOR` but use DescriptorSize instead of `sizeof (EFI_MEMORY_DESCRIPTOR)`.

Reply:
OK. I see.